src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/CngKey.Create.cs

```diff
@@ -0,0 +1,159 @@
+                    if (errorCode != ErrorCode.ERROR_SUCCESS)
+                        throw errorCode.ToCryptographicException();
+                }
+
+                if (creationParameters.KeyUsage.HasValue)
+                {
+                    CngKeyUsages keyUsage = creationParameters.KeyUsage.Value;
+                    ErrorCode errorCode = Interop.NCrypt.NCryptSetProperty(keyHandle, KeyPropertyName.KeyUsage, &keyUsage, sizeof(CngKeyUsages), CngPropertyOptions.Persist);
+                    if (errorCode != ErrorCode.ERROR_SUCCESS)
+                        throw errorCode.ToCryptographicException();
+                }
+
+                if (creationParameters.ParentWindowHandle != IntPtr.Zero)
+                {
+                    IntPtr parentWindowHandle = creationParameters.ParentWindowHandle;
+                    ErrorCode errorCode = Interop.NCrypt.NCryptSetProperty(keyHandle, KeyPropertyName.ParentWindowHandle, &parentWindowHandle, sizeof(IntPtr), CngPropertyOptions.None);
+                    if (errorCode != ErrorCode.ERROR_SUCCESS)
+                        throw errorCode.ToCryptographicException();
+                }
+
+                CngUIPolicy uiPolicy = creationParameters.UIPolicy;
+                if (uiPolicy != null)
+                {
+                    InitializeKeyUiPolicyProperties(keyHandle, uiPolicy);
+                }
+
+                // Iterate over the custom properties, setting those as well.
+                foreach (CngProperty property in creationParameters.Parameters)
+                {
+                    byte[] value = property.GetValueWithoutCopying();
+                    int valueLength = (value == null) ? 0 : value.Length;
+                    fixed (byte* pValue = value.MapZeroLengthArrayToNonNullPointer())
+                    {
+                        ErrorCode errorCode = Interop.NCrypt.NCryptSetProperty(keyHandle, property.Name, pValue, valueLength, property.Options);
+                        if (errorCode != ErrorCode.ERROR_SUCCESS)
+                            throw errorCode.ToCryptographicException();
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        ///     Setup the UIPolicy key properties specified in the key creation parameters
+        /// </summary>
+        private static void InitializeKeyUiPolicyProperties(SafeNCryptHandle keyHandle, CngUIPolicy uiPolicy)
+        {
+            unsafe
+            {
+                fixed (char* pinnedCreationTitle = uiPolicy.CreationTitle, 
+                             pinnedFriendlyName = uiPolicy.FriendlyName,
+                             pinnedDescription = uiPolicy.Description)
+                {
+                    NCRYPT_UI_POLICY ncryptUiPolicy = new NCRYPT_UI_POLICY()
+                    {
+                        dwVersion = 1,
```

Comment:
Is there an associated named constant for version 1?  I feel like Windows is usually pretty good at not requiring magic numbers to float around.